xbmc/games/agents/input/AgentInput.h

```diff
@@ -80,8 +90,10 @@ class CGameAgentManager : public Observable,
   void OnButtonRelease(MOUSE::BUTTON_ID button) override {}
 
   // Public interface
-  GameAgentVec GetAgents() const;
+  std::vector<std::shared_ptr<CAgentController>> GetControllers() const;
```

Comment:
I'm assuming callers shouldn't be able to modify the returned  `CAgentController`s? In that case it would be a good idea to change the return type to `std::vector<std::shared_ptr<const CAgentController>>`. Has the unfortunate side effect that copy in the implementation needs to be explicit:
```C++
std::vector<std::shared_ptr<const CAgentController>> CAgentInput::GetControllers() const
{
  std::lock_guard<std::mutex> lock(m_controllerMutex);
  std::vector<std::shared_ptr<const CAgentController>> controllers(m_controllers.size());
  std::copy(m_controllers.begin(), m_controllers.end(), controllers.begin());
  return controllers;
}
```

Reply:
thanks, updated